src/plugins/data/public/ui/dataset_select/dataset_select.tsx

```diff
@@ -54,19 +54,30 @@ const DatasetSelect: React.FC<DatasetSelectProps> = ({ onSelect, appName }) => {
   } = data;
   const datasetService = queryString.getDatasetService();
 
-  const getTypeFromUri = useCallback((uri?: string): string | undefined => {
-    if (!uri) return undefined;
+  // Helper function to extract data source information from URI format
+  const extractDataSourceInfo = useCallback((uri?: string): { type?: string; name?: string } => {
+    if (!uri) return {};
 
     if (uri.includes('://')) {
       const parts = uri.split('://');
       if (parts.length >= 2) {
-        return parts[0].toUpperCase();
+        const type = parts[0].toUpperCase();
+        const pathParts = parts[1].split('/');
+        const name = pathParts[0];
+        return { type, name };
       }
     }
 
-    return undefined;
+    return { name: uri };
   }, []);
 
+  const getTypeFromUri = useCallback(
+    (uri?: string): string | undefined => {
+      return extractDataSourceInfo(uri).type;
+    },
+    [extractDataSourceInfo]
+  );
```

Comment:
instead of useCallback, i think the functions can be defined outside of the react component

Reply:
True, simply a util function